test/testenv.sh

```diff
@@ -3,9 +3,48 @@
 
 set -e
 
+UNAME=$(uname -s)
+IS_WINDOWS=0
+IS_MAC=0
+if [[ $UNAME == MINGW* || $UNAME == CYGWIN* ]]
+then
+  IS_WINDOWS=1
+elif [[ $UNAME == *Darwin* ]]
+then
+  IS_MAC=1
+fi
+
+resolve_symlink() {
```

Comment:
It's not ideal to have this function in `testenv.sh` but it's simpler to have it available as early as this so that all derived directory vars can start with a resolved version, rather than trying to do it afterwards.